src/main.rs

```diff
@@ -294,6 +341,9 @@ async fn main() -> Result<(), Box<dyn Error>> {
                     }
                 }
             }
+            Some(block) = block_rx.recv() => {
+                node.handle_block(block).await;
```

Comment:
We should do this at the shard level, and not the block level because shards will be making independent progress. And waiting for block confirmation before eviction would be too slow.